### PR TITLE
Add export to cmd bar

### PR DESCRIPTION
**Pull request by franknoirot**
Resolves https://github.com/KittyCAD/modeling-app/issues/847, fixes half of #1497.

- Adds Export as an event on the modeling XState machine
- Allows command bar arguments to define `defaultValue`, `required`, and/or `options` to be functions that take the currently-set argument values as their parameter, unlocking conditional command arguments
- Adds Export as a command with a skippable, conditional `storage` argument

## Demo

https://github.com/KittyCAD/modeling-app/assets/23481541/dad9a299-516c-406d-82bf-fc820c6b842b

### CODE REVIEW

**Review by Irev-Dev — CHANGES_REQUESTED** [review]
Test failures are legitimate.
image

Makes sense, you've removed that button.

A bit of context in case it helps fix the test up. That test is in snapshots because the result is images, what we do is export a model in every format, and then we send the exported images via the CLI back to the engine to get images back with the snapshot endpoint. It's these images we commit to source control. It's convoluted, but it's there because it's a) good to have e2e test for exports since that's an important user flow. and b) we commit the images instead of the exported files themselves because their exact contents should be tested by the export tests in engine land, we just  want to make sure they look right.

So you basically just have to fix up the test to guide it to use the cmd bar instead of the button. As long as exported names stay the same, the cli part of the test should still work.

One wrinkle to look out for, if you search for `playwrightSkipFilePicker` you will see a nasty hack we've done to skip the file picker when in playwright test land. This is because the file picker is OS level and outside of playwrights control so it needs to be skipped. I mention it because in case your changes take you down a different code path and that skip isn't in effect, your going to have a bad time, so make sure the file picker is skipped.

File: src/components/CommandBar/CommandArgOptionInput.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { Combobox } from '@headlessui/react'
 import Fuse from 'fuse.js'
 import { useCommandsContext } from 'hooks/useCommandsContext'
-import { CommandArgumentOption } from 'lib/commandTypes'
-import { useEffect, useRef, useState } from 'react'
+import { CommandArgument, CommandArgumentOption } from 'lib/commandTypes'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
```
Comment (Irev-Dev) [inline src/components/CommandBar/CommandArgOptionInput.tsx:5]:
```suggestion
import { useEffect, useMemo, useRef, useState } from 'react'
```

**Comment by franknoirot** [comment]
Nice, thanks for this sorry for the time suck. Dang I didn't even look at the Playwright snapshot tests because I don't trust them locally and assumed this didn't touch them, that's my bad.

**Comment by franknoirot** [comment]
>Makes sense, you've removed that button.

I didn't remove the button, but I did replace it with a thin button (that just fires the command in the command bar), and I changed its label from "Export Model" to "Export Part". I'm gonna try to get my Ubuntu machine back up and running over the weekend to make sure the tests are good before I re-request

**Comment by Irev-Dev** [comment]
I did look over the code too, I would have approved otherwise, happy for you to merge this one the export test is happy again.

**Comment by Irev-Dev** [comment]
> I'm gonna try to get my Ubuntu machine back up and running over the weekend

FYI the exports rely on api-deux, so in this case you can run only the engine, you will have to run the dev-stack.
I'm pretty sure when I developed this test I just used the live dev endpoint.

Having said that you might be able get away with local not using the dev-stack if you just check the tests is working up until the point where we ask for the export from the backend, and test the final piece with CI. 

Because all that's changed here is the UI to get the export to work has changed, so the test just needs to take different steps to get there, and that half should be do-able with the local engine.

**Review by Irev-Dev — APPROVED** [review]

**Pull request merged**